packages/schema/src/idl.rs

```diff
@@ -0,0 +1,102 @@
+//! The Cosmwasm IDL (Interface Description Language)
+
+use std::collections::BTreeMap;
+
+use schemars::schema::RootSchema;
+use thiserror::Error;
+
+/// The version of the CosmWasm IDL.
+///
+/// Follows Semantic Versioning 2.0.0: https://semver.org/
+///
+/// To determine if a change is breaking, assume consumers allow unknown fields.
+pub const IDL_VERSION: &str = "0.1.0";
+
+/// Rust representation of a contract's API.
+pub struct Api {
+    pub instantiate: RootSchema,
+    pub execute: RootSchema,
+    pub query: RootSchema,
+    pub migrate: Option<RootSchema>,
+    pub sudo: Option<RootSchema>,
+    /// A mapping of query variants to response types
+    pub responses: BTreeMap<String, RootSchema>,
+}
+
+impl Api {
+    pub fn render(self) -> JsonApi<'static> {
+        let mut json_api = JsonApi {
+            contract_name: env!("CARGO_PKG_NAME").to_string(),
```

Comment:
Very cool and what I was asking about.
Surprised that the hackatom used different fields? Maybe an outdated commit???